src/views/AddItem.jsx

```diff
@@ -29,19 +32,42 @@ export function AddItem({ listToken }) {
 
 	// Store values inside itemData and sent over to database
 	// otherwise log an error if request fails
-	const onFormSubmit = (event) => {
-		event.preventDefault();
+	const onFormSubmit = (e) => {
+		e.preventDefault();
 
 		// if list token is not provided, notify user and exit function
 		if (!listToken) {
 			notifyNoToken();
 			return;
 		}
 
+		// We are looking for an array of items that matches the user input
+		// We use `toLowerCase` and `replaceAll` to account for edge cases,
+		// such as casing and extra spaces
+		const nameMatchesArr = data.some((item) => {
+			return (
+				item.name &&
+				item.name.replaceAll(' ', '') ===
+					itemName.toLowerCase().replaceAll(' ', '')
+			);
+		});
+
```

Comment:
Great work getting the the acceptance criteria done! I know that filtering the shopping list with that pesky empty item can be tricky. I think Volha brings up a good point also about punctuation scenarios and got me thinking that maybe a tool we can utilize is [RegEx](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp) or Regular Expression.

This [Reg Ex ](https://www.tutorialspoint.com/regex-in-reactjs)article I found for React.js might be useful and it starts with:
"A RegEx or Regular Expression is a sequence of characters that forms a search pattern and is used to check if a string contains a specified search pattern or not. It is also used to validate strings which consist of email, passwords etc."

Possibly the punctuations can be forbidden in using as an input or triggering a hot toast notification. I may be off base though and offering a more complicated solution than what you both may come up with. But I'm excited to see your updates!

Reply:
OPE Just saw the RegEx commit posted right before I commented this. Feel free to ignore my comment above! lol great work!

Reply:
@alex-andria Thanks Alex! Indeed, I just finished implementing Regex. You have the right idea! And thanks for the articles, I need to brush up on this topic for sure. 👀

Reply:
Lol I thought I missed the regex code here after sending my review and felt silly! But also SAME, I'm not sharp on it either. I've heard general consensus is that developers usually look up premade regex patterns depending on what they need/ I think there's some online tools to help so that we don't need to memorize it. But I need to exercise those as well!

Reply:
@alex-andria Don't feel silly, I love reading your comments! And that is nice to know about the regex tools. I definitely don't want to memorize those sequences. 😆